diff --git a/configure.ac b/configure.ac
index 33c1a7b..bce36a5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -57,8 +57,6 @@ AC_CHECK_HEADERS(gnu/libc-version.h)
 
 AC_CHECK_HEADERS(windows.h,[with_windows=true])
 AM_CONDITIONAL([WITH_WINDOWS], [test "$with_windows" != ""])
-AC_CHECK_TOOL([WINDRES], [windres], [])
-AM_CONDITIONAL([WITH_WIN_ICON], [test "$WINDRES" != ""])
 
 AC_CHECK_MEMBERS([struct dirent.d_type],,,[#include <dirent.h>])
 
diff --git a/src/Makefile.am b/src/Makefile.am
index 02d8629..b0fb5c3 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -35,12 +35,6 @@ ros_SOURCES = main.c opt.c download.c download_windows.c archive.c archive_windo
 	cmd-run-npt.c \
 	cmd-install.c install-sbcl-bin.c install-sbcl-bin_windows.c
 
-if WITH_WIN_ICON
-ros_LDADD += resources.o
-resources.o: resources.rc
-	@WINDRES@ $^ -o $@
-endif
-
 noinst_HEADERS = util.h opt.h cmd-install.h gend.h cmd-run.h
 
 CLEANFILES = gend.h
