diff --git a/config b/config
index 7c6517a..661a777 100644
--- a/config
+++ b/config
@@ -18,6 +18,9 @@ verbose = true
 [patch.'https://github.com/wez/xcb-imdkit-rs']
 xcb-imdkit = { path = "/var/tmp/portage/x11-terms/wezterm-20230712.072601/work/xcb-imdkit-rs-e59e4a6df065b702e0c22fa2dcea00bd8301a446" }
 
+[patch.crates-io]
+xcb = { path = "/var/tmp/portage/x11-terms/wezterm-20230712.072601/work/rust-xcb-dbdaa01c178c6fbe68bd51b7ad44c08172181083" }
+
 [patch.'https://github.com/rust-x-bindings/rust-xcb']
 xcb = { path = "/var/tmp/portage/x11-terms/wezterm-20230712.072601/work/rust-xcb-dbdaa01c178c6fbe68bd51b7ad44c08172181083" }
 
