diff --git a/kwin/main.js.in b/kwin/main.js.in
index 59d7a9f..7d4c7f3 100755
--- a/kwin/main.js.in
+++ b/kwin/main.js.in
@@ -9,7 +9,11 @@ const blacklist = [
     "ksmserver-logout-greeter ksmserver-logout-greeter",
 
     // KDE Plasma splash screen has to be animated only by the login effect.
-    "ksplashqml ksplashqml"
+    "ksplashqml ksplashqml",
+
+    // Spectacle needs to be blacklisted in order to stay out of its own screenshots.
+    "spectacle spectacle", // x11
+    "spectacle org.kde.spectacle" // wayland
 ];
 
 class %EFFECT_CLASS% {
