diff --git a/compiler-shadow.eselect b/compiler-shadow.eselect
index 931a94a..3956270 100644
--- a/compiler-shadow.eselect
+++ b/compiler-shadow.eselect
@@ -106,19 +106,60 @@ do_update() {
 
 		mkdir -p "${out_dir}" || die
 		cd "${out_dir}" || die
-		local p
-		for p in "${wildcards[@]}"; do
-			# remove existing links matching the wildcard
-			rm -f ${p} || die
-
-			# create new links
-			local exe
-			while read -r exe; do
-				ln -v -s "${tool_exec}" "${exe}" || die
-			done < <(compgen -c -X "!${p}" | sort -u)
-			# note: compgen can return a command twice if it is included
-			# in multiple directories in PATH
-		done
+
+		case ${tool_exec} in
+			*sccache)
+				# add rustc to wildcard defaults for sccache
+				[[ ${2} ]] || wildcards+=( 'rustc*' )
+
+				local p
+				for p in "${wildcards[@]}"; do
+					# remove existing wrappers matching the wildcard
+					rm -f ${p} || die
+
+					# remove self wrapper
+					rm -f sccache
+
+					# only use tool basename for sccache
+					tool_exec=$(basename ${tool_exec})
+
+					# create new links
+					local exe
+					while read -r exe; do
+							echo "'${exe}' -> '${tool_exec}'" || die
+							echo "#!/bin/sh" >> "${exe}" || die
+							echo "# make sure we aren't calling ourselves, then run" >> "${exe}" || die
+							echo "PATH=\$(echo \${PATH} | awk -v RS=: -v ORS=: '/${out_dir//\//\\/}/ {next} {print}' | sed 's/:*\$//')" >> "${exe}" || die
+							echo "${tool_exec} ${exe} \"\$@\"" >> "${exe}" || die
+							chmod +x "${exe}" || die
+					done < <(compgen -c -X "!${p}" | sort -u)
+					# note: compgen can return a command twice if it is included
+					# in multiple directories in PATH
+				done
+				# create a self-wrapper to not call sccache on itself
+				echo "'sccache (self)' -> '${tool_exec}'" || die
+				echo "#!/bin/sh" >> "sccache" || die
+				echo "# make sure we aren't calling ourselves, then run" >> "sccache" || die
+				echo "PATH=\$(echo \${PATH} | awk -v RS=: -v ORS=: '/${out_dir//\//\\/}/ {next} {print}' | sed 's/:*\$//')" >> "sccache" || die
+				echo "${tool_exec} \"\$@\"" >> "sccache" || die
+				chmod +x "sccache" || die
+				;;
+			*)
+				local p
+				for p in "${wildcards[@]}"; do
+					# remove existing links matching the wildcard
+					rm -f ${p} || die
+
+					# create new links
+					local exe
+					while read -r exe; do
+							ln -v -s "${tool_exec}" "${exe}" || die
+					done < <(compgen -c -X "!${p}" | sort -u)
+					# note: compgen can return a command twice if it is included
+					# in multiple directories in PATH
+				done
+				;;
+		esac
 	done
 }
 ## }}}
diff --git a/modules/tools/sccache b/modules/tools/sccache
new file mode 100644
index 0000000..8a8c1bf
--- /dev/null
+++ b/modules/tools/sccache
@@ -0,0 +1 @@
+/usr/lib/sccache/bin
