diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 64b262e..828b9f1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -118,11 +118,11 @@ if(NOT APPLE)
     ${QTROOT}/resources
     /usr/share/qt/resources
   )
-  foreach(loc ${rsrc_locations})
-    if(EXISTS ${loc}/qtwebengine_devtools_resources.pak)
-      install(FILES ${loc}/qtwebengine_devtools_resources.pak DESTINATION resources)
-    endif()
-  endforeach()
+#  foreach(loc ${rsrc_locations})
+#    if(EXISTS ${loc}/qtwebengine_devtools_resources.pak)
+#      install(FILES ${loc}/qtwebengine_devtools_resources.pak DESTINATION resources)
+#    endif()
+#  endforeach()
   install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../dist/ DESTINATION ${INSTALL_RESOURCE_DIR}/web-client/desktop)
   install(DIRECTORY ${CMAKE_SOURCE_DIR}/native/ DESTINATION ${INSTALL_RESOURCE_DIR}/web-client/extension)
 endif()
