diff --git a/src/hud_elements.h b/src/hud_elements.h
index 61d1de0..542ac9a 100644
--- a/src/hud_elements.h
+++ b/src/hud_elements.h
@@ -8,6 +8,7 @@
 #include "winesync.h"
 #include "vulkan/vulkan.h"
 #include <array>
+#include <map>
 
 struct Function {
     std::function<void()> run;  // Using std::function instead of a raw function pointer for more flexibility
diff --git a/src/meson.build b/src/meson.build
index 39396d1..eaec2a0 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -159,7 +159,7 @@ if is_unixy
   endif
 endif
 
-link_args = cc.get_supported_link_arguments(['-Wl,-Bsymbolic-functions', '-Wl,-z,relro', '-Wl,--exclude-libs,ALL', '-lGL', '-static-libstdc++'])
+link_args = cc.get_supported_link_arguments(['-Wl,-Bsymbolic-functions', '-Wl,-z,relro', '-Wl,--exclude-libs,ALL', '-lGL'])
 # meson fails to check version-script so just force add
 link_args += '-Wl,--version-script,@0@'.format(join_paths(meson.current_source_dir(), 'mangohud.version'))
 
