From 12a86ef36e308e281f4f1bc4b3ac5613e24ad53c Mon Sep 17 00:00:00 2001
From: FlightlessMango <flightlessmangoyt@gmail.com>
Date: Fri, 5 May 2023 02:29:14 +0200
Subject: [PATCH] meson: prefer system cmocka

---
 meson.build | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index a32d04dc9a..dd936c3c99 100644
--- a/meson.build
+++ b/meson.build
@@ -250,8 +250,7 @@ if get_option('include_doc')
 endif
 
 if get_option('tests').enabled()
-  cmocka = subproject('cmocka')
-  cmocka_dep = cmocka.get_variable('cmocka_dep')
+  cmocka_dep = dependency('cmocka', fallback: ['cmocka', 'cmocka_dep'])
 
   e = executable('amdgpu', 'tests/test_amdgpu.cpp',
     files(
