diff --git a/bin/iwd-firewalld-zone b/bin/iwd-firewalld-zone
index 257110e..4a23430 100755
--- a/bin/iwd-firewalld-zone
+++ b/bin/iwd-firewalld-zone
@@ -105,7 +105,8 @@ function trim_whitespace() {
 
 function get_wanted_zone() {
 	local network="${1}"
-	awk -F= 'BEGIN{rc=1} $1=="FirewalldZone" {print; rc=0; exit} END{exit rc}' "$(config_file "${1}")" 2>/dev/null
+	local setting="$(awk -F= 'BEGIN{rc=1} $1=="FirewalldZone" {print; rc=0; exit} END{exit rc}' "$(config_file "${1}")" 2>/dev/null)"
+	echo "${setting#*=}"
 }
 
 
